### PR TITLE
Increase unit test code coverage of `ble/` by 1.9%

### LINKED ISSUE #37232: Increase unit test code coverage of `ble/` to 80%
Reference:  https://github.com/project-chip/connectedhomeip/issues/36537

### PR DESCRIPTION
**Pull request by AniDashyan**
#### Summary

This PR adds comprehensive unit tests for the `BleLayer::NewConnectionByDiscriminators` method in `TestBleLayer.cpp`, increasing  test coverage of `ble/` folder from 67.2% to 69.1%. The tests cover the following scenarios:
* Connection attempt with an uninitialized `BleLayer`
* Connection attempt without a configured `BleConnectionDelegate`
* Connection attempt with a null BLE transport
* Successful connection establishment (via `OnSuccess` callback)
* Connection failure handling (via `OnError` callback)
* Edge case with an empty discriminator list

#### Implementation Notes
* Tests are limited to single discriminator scenarios due to the `BLE_LAYER_NUM_BLE_ENDPOINTS` macro being set to 1, restricting `BleLayer` to one connection at a time. Multiple `OnSuccess` callback calls when multiple discriminators match cannot be tested at the moment beacuse of this endpoint limitation.
* Created `BleLayerTestAccess` accessor class in  `BleLayerTestAccess.h` to access private member `mConnectionDelegate` and `OnConnectionComplete` method of `BleLayer` for testing purposes, following the testing guidelines defined in the [Connected Home IP unit testing documentation](https://project-chip.github.io/connectedhomeip-doc/testing/unit_testing.html)
* `OnSuccess` and `OnError` callback validation is achieved through custom lambda functions implemented specifically for testing purposes to simulate connection success/failure scenarios.

#### Related issues
Main Issue [#37732](https://github.com/project-chip/connectedhomeip/issues/37232)

#### Testing

This PR only adds new unit tests. No changes made to production code.

### CODE REVIEW

File: src/ble/tests/BUILD.gn [hunk 1]
```diff
@@ -27,6 +27,8 @@ chip_test_suite("tests") {
     "TestBtpEngine.cpp",
   ]
 
+  sources = [ "BleLayerTestAccess.h" ]
```
[the remark this answers is not in the document]
Comment (AniDashyan) [reply src/ble/tests/BUILD.gn:30]: The list of existing source files is in `test_sources`, meanwhile the TestAccess header files are added in `sources` as it is defined in [unit testing documentation](https://project-chip.github.io/connectedhomeip-doc/testing/unit_testing.html)

File: src/ble/BleLayer.h [hunk 2]
```diff
@@ -224,7 +231,7 @@ class DLL_EXPORT BleLayer
         kState_NotInitialized = 0,
         kState_Initialized    = 1,
         kState_Disconnecting  = 2
-    } mState; ///< [READ-ONLY] external access is deprecated, use IsInitialized() / IsBleClosing()
+    } mState; ///< [READ-ONLY] external access is deprecated, use  () / IsBleClosing()
```
Comment (bzbarsky-apple) [inline src/ble/BleLayer.h]: Why was the bot's (correct) comment resolved without actually resolving the issue?

File: src/ble/BleLayer.h [hunk 3]
```diff
@@ -354,6 +361,7 @@ class DLL_EXPORT BleLayer
 
     static void OnConnectionComplete(void * appState, BLE_CONNECTION_OBJECT connObj);
     static void OnConnectionError(void * appState, CHIP_ERROR err);
+
```
Comment (bzbarsky-apple) [inline src/ble/BleLayer.h]:
```suggestion
```

File: src/ble/tests/TestBleLayer.cpp [hunk 4]
```diff
@@ -383,5 +408,127 @@ TEST_F(TestBleLayer, ExceedBleConnectionEndPointLimit)
...
 }
 
+TEST_F(TestBleLayer, NewBleConnectionByDiscriminatorsNotInitialized)
+{
+    // Simulate BleLayer not being initialized by calling Shutdown
+    Shutdown();
+
+    // Create a list of discriminators
+    SetupDiscriminator discriminators[] = { SetupDiscriminator(), SetupDiscriminator() };
+    Span<const SetupDiscriminator> discriminatorsSpan(discriminators, 2);
```
Comment (bzbarsky-apple) [inline src/ble/tests/TestBleLayer.cpp]:
```suggestion
    Span<const SetupDiscriminator> discriminatorsSpan(discriminators);
```

File: src/ble/tests/TestBleLayer.cpp [hunk 5]
```diff
@@ -383,5 +408,127 @@ TEST_F(TestBleLayer, ExceedBleConnectionEndPointLimit)
...
+
+    // Verify that the success callback was called
+    EXPECT_EQ(mOnConnectionCompleteCount, 1);
+    EXPECT_EQ(mOnConnectionErrorCount, 0);
+}
+
+// Checks that the connection could not be established due to an error
+TEST_F(TestBleLayer, NewConnectionByDiscriminatorsError)
+{
+    chip::Test::BleLayerTestAccess access(this);
```
Comment (bzbarsky-apple) [inline src/ble/tests/TestBleLayer.cpp:522]: Why the `chip::` prefix here and elsewhere in this PR?  Is this not all inside `namespace chip`?

File: src/ble/tests/TestBleLayer.cpp [hunk 6]
```diff
@@ -383,5 +408,127 @@ TEST_F(TestBleLayer, ExceedBleConnectionEndPointLimit)
...
+}
+
+TEST_F(TestBleLayer, NewBleConnectionByDiscriminatorsNoConnectionDelegate)
+{
+    // Set up the BleLayerTestAccess accessor class to manipulate the BleConnectionDelegate of BleLayer
+    chip::Test::BleLayerTestAccess access(this);
+    access.SetConnectionDelegate(nullptr);
+
+    SetupDiscriminator discriminators[] = { SetupDiscriminator(), SetupDiscriminator() };
+    Span<const SetupDiscriminator> discriminatorsSpan(discriminators, 2);
```
Comment (bzbarsky-apple) [inline src/ble/tests/TestBleLayer.cpp]: Please drop the extra `, 2` various places.  I'll stop commenting on this.

Comment (AniDashyan) [reply src/ble/tests/TestBleLayer.cpp]: Thank you for the review. This PR is still a draft, I'll make the changes based on your feedback.

File: src/ble/tests/TestBleLayer.cpp (hunk 4 above)
Comment (AniDashyan) [reply src/ble/tests/TestBleLayer.cpp]: Done

File: src/ble/tests/TestBleLayer.cpp (hunk 5 above)
Comment (AniDashyan) [reply src/ble/tests/TestBleLayer.cpp:522]: The `chip::` prefix here is needed, because otherwise compiler identifies `Test` as `pw::unit_test::internal::Test`.

File: src/ble/BleLayer.h (hunk 3 above)
Comment (AniDashyan) [reply src/ble/BleLayer.h]: Done. Thank you.

File: src/ble/tests/TestBleLayer.cpp [hunk 7]
```diff
@@ -383,5 +408,127 @@
...
+
+    EXPECT_EQ(NewBleConnectionByDiscriminators(discriminatorsSpan, this, OnSuccess, OnError), CHIP_NO_ERROR);
+
+    // Call the error callback directly to simulate an error
+    OnError(bleLayerState, CHIP_ERROR_CONNECTION_ABORTED);
+
+    // Verify that the error callback was called
+    EXPECT_EQ(mOnConnectionCompleteCount, 0);
+    EXPECT_EQ(mOnConnectionErrorCount, 1);
+}
```
[the remark this answers is not in the document]
Comment (AniDashyan) [reply src/ble/tests/TestBleLayer.cpp]: The test is implemented this way because:

1. `BleConnectionDelegate::NewConnection` has no implementation in tests (it's platform-specific). Calling `NewBleConnectionByDiscriminators` would just hit this empty method and never trigger callbacks. 
The suggested alternative won't work - calling `NewBleConnectionByDiscriminators` and waiting for callbacks would fail because the delegate is empty and never calls back.
2. Calling the lambdas directly is the only way to test the callback handling logic, since the real delegate would never call back in a unit test environment.
3. `CallOnConnectionComplete()` is a wrapper for the actual private method that handles connection completion.

File: src/ble/tests/TestBleLayer.cpp [hunk 8]
```diff
@@ -383,5 +408,127 @@ TEST_F(TestBleLayer, ExceedBleConnectionEndPointLimit)
...
+
+    EXPECT_EQ(NewBleConnectionByDiscriminators(discriminatorsSpan, this, OnSuccess, OnError), CHIP_NO_ERROR);
+
+    // Simulate a successful connection by calling the success callback directly
+    OnSuccess(bleLayerState, discriminatorsSpan[0].GetLongValue(), GetConnectionObject());
+
+    // Verify that the success callback was called
+    EXPECT_EQ(mOnConnectionCompleteCount, 1);
+    EXPECT_EQ(mOnConnectionErrorCount, 0);
+}
```
[the remark this answers is not in the document]
Comment (AniDashyan) [reply src/ble/tests/TestBleLayer.cpp]: The test is implemented this way because:

1. `BleConnectionDelegate::NewConnection` has no implementation in tests (it's platform-specific). Calling `NewBleConnectionByDiscriminators` would just hit this empty method and never trigger callbacks. The suggested alternative won't work - calling `NewBleConnectionByDiscriminators` and waiting for callbacks would fail because the delegate is empty and never calls back.
2. Calling the lambdas directly is the only way to test the callback handling logic, since the real delegate would never call back in a unit test environment.
3. `CallOnConnectionComplete()` is a wrapper for the actual private method that handles connection completion.

File: src/ble/tests/TestBleLayer.cpp (hunk 7 above)
[the remark this answers is not in the document]
Comment (AniDashyan) [reply src/ble/tests/TestBleLayer.cpp]: Same reasoning as the previous test:
1. `The BleConnectionDelegate` implementation is empty (platform-specific), so calling `NewBleConnectionByDiscriminators` would never trigger any callbacks. Direct lambda invocation is the only way to test the error handling logic in a u nit test environment. The suggested method fails because the delegate has no implementation to trigger the callbacks we need to test.
2. `CallOnConnectionError()` is the correct wrapper for `BleLayer::OnConnectionError()` - the actual private method that handles connection errors.

**Review by heinsaar — APPROVED** [review]

File: src/ble/tests/TestBleLayer.cpp [hunk 9]
```diff
@@ -383,5 +408,127 @@ TEST_F(TestBleLayer, ExceedBleConnectionEndPointLimit)
     EXPECT_FALSE(HandleWriteReceivedCapabilitiesRequest(connObj));
 }
 
+TEST_F(TestBleLayer, NewBleConnectionByDiscriminatorsNotInitialized)
```
Comment (juandediosg) [inline src/ble/tests/TestBleLayer.cpp:444]: Maybe consider adding a comment.

File: src/ble/tests/TestBleLayer.cpp [hunk 10]
```diff
@@ -383,5 +408,127 @@ TEST_F(TestBleLayer, ExceedBleConnectionEndPointLimit)
...
+    Span<const SetupDiscriminator> discriminatorsSpan(discriminators);
+
+    // Define success and error callbacks
+    auto OnSuccess = [](void * appState, uint16_t matchedLongDiscriminator, BLE_CONNECTION_OBJECT connObj) {};
+    auto OnError   = [](void * appState, CHIP_ERROR err) {};
+
+    EXPECT_EQ(NewBleConnectionByDiscriminators(discriminatorsSpan, this, OnSuccess, OnError), CHIP_ERROR_INCORRECT_STATE);
+}
+
+TEST_F(TestBleLayer, NewBleConnectionByDiscriminatorsNoConnectionDelegate)
```
Comment (juandediosg) [inline src/ble/tests/TestBleLayer.cpp:461]: Maybe consider adding a comment.

File: src/ble/tests/TestBleLayer.cpp [hunk 11]
```diff
@@ -383,5 +408,127 @@ TEST_F(TestBleLayer, ExceedBleConnectionEndPointLimit)
...
+    SetupDiscriminator discriminators[] = { SetupDiscriminator(), SetupDiscriminator() };
+    Span<const SetupDiscriminator> discriminatorsSpan(discriminators);
+
+    auto OnSuccess = [](void * appState, uint16_t matchedLongDiscriminator, BLE_CONNECTION_OBJECT connObj) {};
+    auto OnError   = [](void * appState, CHIP_ERROR err) {};
+
+    EXPECT_EQ(NewBleConnectionByDiscriminators(discriminatorsSpan, this, OnSuccess, OnError), CHIP_ERROR_INCORRECT_STATE);
+}
+
+TEST_F(TestBleLayer, NewBleConnectionByDiscriminatorsNoBleTransportLayer)
```
Comment (juandediosg) [inline src/ble/tests/TestBleLayer.cpp:477]: Maybe consider adding a comment.

File: src/ble/tests/TestBleLayer.cpp [hunk 12]
```diff
@@ -383,5 +408,127 @@ TEST_F(TestBleLayer, ExceedBleConnectionEndPointLimit)
...
+    SetupDiscriminator discriminators[] = { SetupDiscriminator(), SetupDiscriminator() };
+    Span<const SetupDiscriminator> discriminatorsSpan(discriminators);
+
+    auto OnSuccess = [](void * appState, uint16_t matchedLongDiscriminator, BLE_CONNECTION_OBJECT connObj) {};
+    auto OnError   = [](void * appState, CHIP_ERROR err) {};
+
+    EXPECT_EQ(NewBleConnectionByDiscriminators(discriminatorsSpan, this, OnSuccess, OnError), CHIP_ERROR_INCORRECT_STATE);
+}
+
+TEST_F(TestBleLayer, NewConnectionByDiscriminatorsSuccess)
```
Comment (juandediosg) [inline src/ble/tests/TestBleLayer.cpp:491]: Maybe consider adding a comment.

File: src/ble/tests/TestBleLayer.cpp [hunk 13]
```diff
@@ -383,5 +408,127 @@ TEST_F(TestBleLayer, ExceedBleConnectionEndPointLimit)
...
+
+    // Call the error callback directly to simulate an error
+    OnError(bleLayerState, CHIP_ERROR_CONNECTION_ABORTED);
+
+    // Verify that the error callback was called
+    EXPECT_EQ(mOnConnectionCompleteCount, 0);
+    EXPECT_EQ(mOnConnectionErrorCount, 1);
+}
+
+TEST_F(TestBleLayer, NewConnectionByDiscriminatorsEmptySpan)
```
Comment (juandediosg) [inline src/ble/tests/TestBleLayer.cpp:553]: Maybe consider adding a comment.

File: src/ble/BleLayer.h (hunk 2 above)
Comment (AniDashyan) [reply src/ble/BleLayer.h]: Thank you. Done.

File: src/ble/tests/TestBleLayer.cpp (hunk 9 above)
Comment (AniDashyan) [reply src/ble/tests/TestBleLayer.cpp:444]: Thank you. Done.

File: src/ble/tests/TestBleLayer.cpp (hunk 10 above)
Comment (AniDashyan) [reply src/ble/tests/TestBleLayer.cpp:461]: Done.

File: src/ble/tests/TestBleLayer.cpp (hunk 12 above)
Comment (AniDashyan) [reply src/ble/tests/TestBleLayer.cpp:491]: Done.

**Review by juandediosg — APPROVED** [review]

File: src/ble/tests/BleLayerTestAccess.h [hunk 14]
```diff
@@ -0,0 +1,49 @@
...
+namespace Test {
+/**
+ * @brief Class acts as an accessor to private methods of the BleLayer class without needing to give friend access to
+ *        each individual test.
+ */
+class BleLayerTestAccess
+{
+public:
+    BleLayerTestAccess() = delete;
+    BleLayerTestAccess(Ble::BleLayer * layer) : mLayer(layer) {}
```
[the remark this answers is not in the document]
Comment (AniDashyan) [reply src/ble/tests/BleLayerTestAccess.h:33]: `mLayer` can't be a reference because in  `TestBleLayer.cpp` file it takes `this` as a value.

File: src/ble/tests/TestBleLayer.cpp [hunk 15]
```diff
@@ -383,5 +408,132 @@
...
+    Span<const SetupDiscriminator> discriminatorsSpan(discriminators);
+
+    auto OnSuccess = [](void * appState, uint16_t matchedLongDiscriminator, BLE_CONNECTION_OBJECT connObj) {
+        BleLayer * testLayer = static_cast<BleLayer *>(appState);
+        chip::Test::BleLayerTestAccess tempAccess(testLayer);
+
+        tempAccess.CallOnConnectionComplete(appState, connObj);
+    };
+    auto OnError             = [](void * appState, CHIP_ERROR err) { FAIL() << "OnError should not be called in this test"; };
+    BleLayer * bleLayerState = this;
```
[the remark this answers is not in the document]
Comment (AniDashyan) [reply src/ble/tests/TestBleLayer.cpp:507]: `bleLayerState` variable  is not redundant because direct use of 'this' in `OnSuccess` function causes pointer offset issues when cast to `void*`  and back to `BleLayer*` in the callback, leading to segfaults.

File: src/ble/tests/TestBleLayer.cpp [hunk 16]
```diff
@@ -383,5 +408,132 @@
...
+        FAIL() << "OnSuccess should not be called in this test";
+    };
+
+    auto OnError = [](void * appState, CHIP_ERROR err) {
+        BleLayer * testLayer = static_cast<BleLayer *>(appState);
+        chip::Test::BleLayerTestAccess tempAccess(testLayer);
+
+        tempAccess.CallOnConnectionError(appState, err);
+    };
+    BleLayer * bleLayerState = this;
```
[the remark this answers is not in the document]
Comment (AniDashyan) [reply src/ble/tests/TestBleLayer.cpp]: `bleLayerState` variable is not redundant because direct use of 'this' in `OnError` function causes pointer offset issues when cast to `void*` and back to `BleLayer*` in the callback, leading to segfaults.

File: src/ble/tests/TestBleLayer.cpp [hunk 17]
```diff
@@ -383,5 +408,132 @@ TEST_F(TestBleLayer, ExceedBleConnectionEndPointLimit)
...
+    SetupDiscriminator discriminators[] = { SetupDiscriminator(), SetupDiscriminator() };
+    Span<const SetupDiscriminator> discriminatorsSpan(discriminators);
+
+    auto OnSuccess = [](void * appState, uint16_t matchedLongDiscriminator, BLE_CONNECTION_OBJECT connObj) {};
+    auto OnError   = [](void * appState, CHIP_ERROR err) {};
+
+    EXPECT_EQ(NewBleConnectionByDiscriminators(discriminatorsSpan, this, OnSuccess, OnError), CHIP_ERROR_INCORRECT_STATE);
+}
+
+// Simulate successfull connection callback from delegate
```
[the remark this answers is not in the document]
Comment (AniDashyan) [reply src/ble/tests/TestBleLayer.cpp]: Done

File: src/ble/tests/BleLayerTestAccess.h [hunk 18]
```diff
@@ -0,0 +1,49 @@
...
+    // Wrapper for OnConnectionComplete private method
+    void CallOnConnectionComplete(void * appState, BLE_CONNECTION_OBJECT connObj)
+    {
+        mLayer->OnConnectionComplete(appState, connObj);
+    }
+
+    void CallOnConnectionError(void * appState, CHIP_ERROR err) { mLayer->OnConnectionError(appState, err); }
+
+private:
+    Ble::BleLayer * mLayer = nullptr;
```
[the remark this answers is not in the document]
Comment (AniDashyan) [reply src/ble/tests/BleLayerTestAccess.h:46]: Same reason as above.

File: src/ble/tests/TestBleLayer.cpp [hunk 19]
```diff
@@ -383,5 +408,132 @@
...
+{
+    chip::Test::BleLayerTestAccess access(this);
+    access.SetConnectionDelegate(this);
+
+    Span<const SetupDiscriminator> discriminatorsSpan;
+
+    auto OnSuccess = [](void * appState, uint16_t matchedLongDiscriminator, BLE_CONNECTION_OBJECT connObj) {};
+    auto OnError   = [](void * appState, CHIP_ERROR err) {};
+
+    EXPECT_NE(NewBleConnectionByDiscriminators(discriminatorsSpan, this, OnSuccess, OnError), CHIP_NO_ERROR);
```
[the remark this answers is not in the document]
Comment (AniDashyan) [reply src/ble/tests/TestBleLayer.cpp]: Done

File: src/ble/BleLayer.h [hunk 20]
```diff
@@ -224,7 +230,7 @@ class DLL_EXPORT BleLayer
         kState_NotInitialized = 0,
         kState_Initialized    = 1,
         kState_Disconnecting  = 2
-    } mState; ///< [READ-ONLY] external access is deprecated, use IsInitialized() / IsBleClosing()
+    } mState; ///< [READ-ONLY] external access is deprecated, use  IsInitialized() / IsBleClosing()
```
Comment (andy31415) [inline src/ble/BleLayer.h]:
```suggestion
    } mState; ///< [READ-ONLY] external access is deprecated, use IsInitialized() / IsBleClosing()
```

File: src/ble/tests/TestBleLayer.cpp [hunk 21]
```diff
@@ -383,5 +408,132 @@ TEST_F(TestBleLayer, ExceedBleConnectionEndPointLimit)
...
+    auto OnSuccess = [](void * appState, uint16_t matchedLongDiscriminator, BLE_CONNECTION_OBJECT connObj) {};
+    auto OnError   = [](void * appState, CHIP_ERROR err) {};
+
+    EXPECT_EQ(NewBleConnectionByDiscriminators(discriminatorsSpan, this, OnSuccess, OnError), CHIP_ERROR_INCORRECT_STATE);
+}
+
+// Verify connection fails when Ble Transport Layer is missing
+TEST_F(TestBleLayer, NewBleConnectionByDiscriminatorsNoBleTransportLayer)
+{
+    chip::Test::BleLayerTestAccess access(this);
```
Comment (andy31415) [inline src/ble/tests/TestBleLayer.cpp]: since the test access is on `this`, does this not mean that we are already on `this` and have access to all public/private methods?

Why do we use BleLayerTestAccess in this case?

Same comment throughout

**Review by andy31415 — APPROVED** [review]
Seems ok, except the question on why we have the test access bit.

File: src/ble/BleLayer.h (hunk 20 above)
Comment (AniDashyan) [reply src/ble/BleLayer.h]: Done.

File: src/ble/tests/BUILD.gn (hunk 1 above)
[the remark this answers is not in the document]
Comment (AniDashyan) [reply src/ble/tests/BUILD.gn:30]: `TestAccess` header files are added in `sources` as it is defined in [unit testing documentation](https://project-chip.github.io/connectedhomeip-doc/testing/unit_testing.html)

File: src/ble/tests/TestBleLayer.cpp [hunk 22]
```diff
@@ -383,5 +408,132 @@ TEST_F(TestBleLayer, ExceedBleConnectionEndPointLimit)
...
 
+// Verify connection attempt fails when BleLayer is unitialized
+TEST_F(TestBleLayer, NewBleConnectionByDiscriminatorsNotInitialized)
+{
+    // Simulate BleLayer not being initialized by calling Shutdown
+    Shutdown();
+
+    // Create a list of discriminators
+    SetupDiscriminator discriminators[] = { SetupDiscriminator(), SetupDiscriminator() };
+    Span<const SetupDiscriminator> discriminatorsSpan(discriminators);
```
[the remark this answers is not in the document]
Comment (AniDashyan) [reply src/ble/tests/TestBleLayer.cpp]: Used a single discriminator for clarity.

File: src/ble/tests/TestBleLayer.cpp [hunk 23]
```diff
@@ -383,5 +408,132 @@ TEST_F(TestBleLayer, ExceedBleConnectionEndPointLimit)
...
+
+// Verify connection attempt fails when there is no BleConnectionDelegate
+TEST_F(TestBleLayer, NewBleConnectionByDiscriminatorsNoConnectionDelegate)
+{
+    // Set up the BleLayerTestAccess accessor class to manipulate the BleConnectionDelegate of BleLayer
+    chip::Test::BleLayerTestAccess access(this);
+    access.SetConnectionDelegate(nullptr);
+
+    SetupDiscriminator discriminators[] = { SetupDiscriminator(), SetupDiscriminator() };
+    Span<const SetupDiscriminator> discriminatorsSpan(discriminators);
```
[the remark this answers is not in the document]
Comment (AniDashyan) [reply src/ble/tests/TestBleLayer.cpp]: Used a single discriminator for clarity.

File: src/ble/tests/TestBleLayer.cpp (hunk 21 above)
Comment (AniDashyan) [reply src/ble/tests/TestBleLayer.cpp]: We use `BleLayerTestAccess` because we need to test private members of `BleLayer`. Even though `TestBleLayer` inherits from `BleLayer`, `mConnectionDelegate` is a private member, so it can't be accessed directly through `this`. To test what happens when BLE Transport is missing, we need the check `VerifyOrError(mConnectionDelegate != nullptr, CHIP_ERROR_INCORRECT_STATE)` to pass first in `NewBleConnectionByDiscriminators` function. But `mConnectionDelegate` starts as nullptr, so we have to set it to something else. Since `mConnectionDelegate` is private, we can't change it directly through `this` - that's why we need the accessor class.
